src/screens/chat/index.js

```diff
@@ -1,40 +1,58 @@
 import React, { PureComponent } from 'react';
 import { GiftedChat } from 'react-native-gifted-chat';
 import { connect } from 'react-redux';
-import { openChat, sendMessage } from '../../store/socket';
+import { sendMessage } from '../../store/socket';
+
 
 
 
 class Chat extends PureComponent {
-    constructor(props) {
-        super(props);
+    state = {
+        messages: []
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (state.messages !== props.messages) {
+            return {
+                messages: props.messages,
+            };
+        }
+        return null;
     }
 
-    // componentDidMount() {
-    //     openChat({
-    //         user: this.props.user, receiver: this.props.receiver
-    //     });
-    // }
 
     handleSend = (message) => {
-        sendMessage(message.text);
+        const { chat, user } = this.props
+
+        sendMessage({
+            user: {
+                _id: user._id,
+                name: user.name
+            },
+            chatId: chat._id,
+            text: message.text,
+            createdAt: message.createdAt
+        });
     }
 
     render() {
+        const { user } = this.props
+        const { messages } = this.state
+
         return (
             <GiftedChat
-                messages={this.props.messages}
+                messages={messages}
                 user={{
-                    _id: 1, //this.props.user.id
+                    _id: user._id
                 }}
-                onSend={message => this.handleSend(message[0])}
+                onSend={messages => this.handleSend(messages[0])}
             />
         );
     }
 }
 
-export default connect((state, { navigation }) => ({
-    messages: state.messages,
+export default connect((state) => ({
```

Comment:
Brackets here is redundantly
`export default connect( state => ({`

Reply:
8.4 If your function takes a single argument and doesn’t use braces, omit the parentheses. Otherwise, always include parentheses around arguments for clarity and consistency. Note: it is also acceptable to always use parentheses, in which case use the “always” option for eslint. eslint: arrow-parens

Reply:
You can use braces for one argument function, but i and Eugene prefer without braces. As i see this is important for you, I will no longer comment on this. 
But  **one-liners** transform into **blocks of code** it's redundantly code for me and I will always comment on them